aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -307,8 +328,12 @@ module aptos_framework::fungible_asset {
 
     #[view]
     /// Get the balance of a given store.
-    public fun balance<T: key>(store: Object<T>): u64 acquires FungibleStore {
-        if (store_exists(object::object_address(&store))) {
+    public fun balance<T: key>(store: Object<T>): u64 acquires FungibleStore, ConcurrentFungibleBalance {
+        let store_addr = object::object_address(&store);
+        if (concurrent_fungible_balance_exists(store_addr)) {
```

Comment:
We should check here that `&& store_exists(store_addr)` as well?

Reply:
This can be part of `concurrent_fungible_balance_exists` function?

Reply:
that should be a general invariant, but not sure we want to/need to explicitly check it here?